File: src/main/java/com/patres/languagepopup/gui/controller/model/TextActionController.java
package com.patres.languagepopup.gui.controller.model;

import com.jfoenix.controls.JFXButton;
import com.jfoenix.controls.JFXTextField;
import com.patres.languagepopup.Main;
import com.patres.languagepopup.gui.controller.Controllable;
import com.patres.languagepopup.gui.controller.PointerController;
import javafx.fxml.FXML;
import javafx.scene.Scene;
import javafx.scene.control.Label;
import javafx.stage.Stage;
import javafx.stage.StageStyle;

public class TextActionController extends AutomationController {

    @FXML
    private JFXTextField valueTextField;
    @FXML
    private Label actionLabel;
    @FXML
    private Label validLabel;
    @FXML
    private JFXButton pointButton;
    @FXML
    private JFXButton removeButton;

    @FXML
    public void initialize() {
        setHandler();
    }

    private void setHandler() {
        pointButton.setOnAction((event) -> {
            Main.Companion.getMainStage().setIconified(true);
            showPointerStage();
        });
    }

    private void showPointerStage() {
        Stage stage = new Stage();
        stage.initStyle(StageStyle.TRANSPARENT);
        stage.setTitle("Set point");
        stage.setScene(loadPointScene(stage));
        stage.setMaximized(true);
        stage.show();
    }

    private Scene loadPointScene(Stage stage) {
        PointerController pointController = new PointerController(stage, this);
        return pointController.getScene();
    }

    public void showButton(boolean show) {
        pointButton.setVisible(show);
    }

    public String getValue() {
        return valueTextField.getText();
    }

    public Label getActionLabel() {
        return actionLabel;
    }

    public JFXTextField getValueTextField() {
        return valueTextField;
    }

    public Label getValidLabel() {
        return validLabel;
    }
}

File: src/main/java/com/patres/languagepopup/gui/controller/model/TextActionController.kt
package com.patres.languagepopup.gui.controller.model

import com.jfoenix.controls.JFXButton
import com.jfoenix.controls.JFXTextField
import com.patres.languagepopup.Main
import com.patres.languagepopup.gui.controller.PointerController
import com.patres.languagepopup.model.AutomationModel
import com.patres.languagepopup.model.SchemaGroupModel
import com.patres.languagepopup.model.TextActionModel
import javafx.fxml.FXML
import javafx.scene.Node
import javafx.scene.Scene
import javafx.scene.control.Label
import javafx.scene.layout.BorderPane
import javafx.scene.layout.Pane
import javafx.stage.Stage
import javafx.stage.StageStyle

class TextActionController : AutomationController() {

    @FXML
    lateinit var mainPane: BorderPane

    @FXML
    lateinit var valueTextField: JFXTextField

    @FXML
    lateinit var actionLabel: Label

    @FXML
    lateinit var validLabel: Label

    @FXML
    lateinit var pointButton: JFXButton


    lateinit var model: TextActionModel

    override fun getModel(): AutomationModel<out AutomationController> = model

    val value: String
        get() = valueTextField.text


    @FXML
    fun initialize() {
        setHandler()
    }

    private fun setHandler() {
        pointButton.setOnAction {
            Main.mainStage.isIconified = true
            showPointerStage()
        }
    }

    private fun showPointerStage() {
        val stage = Stage()
        stage.initStyle(StageStyle.TRANSPARENT)
        stage.title = "Set point"
        stage.scene = loadPointScene(stage)
        stage.isMaximized = true
        stage.show()
    }

    private fun loadPointScene(stage: Stage): Scene {
        val pointController = PointerController(stage, this)
        return pointController.scene
    }

    fun getMainOutsideNode(): Node = mainPane

    fun getMainInsideNode(): Pane = mainPane
}
